app/Services/MentionsService.php

```diff
@@ -568,13 +593,28 @@ protected function parseAttributesForEdit(): self
      */
     protected function entity(int $id): Entity|null
     {
-        if (!Arr::has($this->entities, (string) $id)) {
+        if (!Arr::has($this->entities, (string) $id) && !Arr::has($this->privateEntities, (string) $id)) {
             $this->entities[$id] = Entity::where(['id' => $id])->first();
         }
 
         return Arr::get($this->entities, $id);
     }
 
+    /**
+     */
+    protected function hiddenEntity(int $id): Entity|null
+    {
+        if (!$this->campaign->getNamePrivateEntities()) {
+            return null;
+        }
+
+        if (!Arr::has($this->entities, (string) $id) && !Arr::has($this->privateEntities, (string) $id)) {
+            $this->privateEntities[$id] = Entity::where(['id' => $id])->withHidden()->first();
```

Comment:
Can I mention entities from another campaign this way? If so, we need a `['campaign_id' => $this->campaign->id]` in the where